### DISCUSSION TITLE
Extending THM FlowJunction problem

### DISCUSSION THREAD

**Discussion by SomnusYu** [idaholab/moose#25596] [Q&A Modules: General]
Hi all,

I noticed that FlowJuction only has [FlowJunction1Phase] for single-phase flow. And the variables of _VolumeJunction1Phase_ are **{rhoV, rhouV, rhovV, rhowV, rhoEV}**.
If I would like to extend this model to a two-phase flow using the six-equation model (variables are α, p, ul, ug, Tl, Tg), what the variables of _VolumeJunction2Phase_ should be? Should it be **{αl_rhol_V, αg_rhog_V, αl_rhol_ul_V, αl_rhol_vl_V, αl_rhol_wl_V, αg_rhog_ug_V, αg_rhog_vg_V, αg_rhog_wg_V, αl_rhol_El_V, αg_rhog_Eg_V}**?

Thank you very much!

**Comment by lindsayad** [comment]
@joshuahansel @licharlot. I know that the existing two phase extension of THM, Relap-7, uses a 7 equation model

**Reply by licharlot** [reply]
The choice of the variables is up to the developper for a 2 phase flow extension, but the set of variables above is the natural extension of the single phase version. Note that THM is using rhoA, rhouA and rhoEA as variables, not (p, u, T)

**Reply by SomnusYu** [reply]
Thank you for your reply! Yes, the variables depend on the developer. What I would like to know is the scalar kernel of the 2-phase flow in VolumeJunction2Phase not in other regions. 

Just like VolumeJunction1Phase, the scalar kernel is
![image](https://github.com/idaholab/moose/assets/118275988/80a8da47-8ee4-41ab-a2e2-022e212bc660), which using **{rhoV, rhouV, rhovV, rhowV, rhoEV}** as scalar variables. 

So I think the scalar variables of 2-phase flow in VolumeJunction2Phase should be **{αl_rhol_V, αg_rhog_V, αl_rhol_ul_V, αl_rhol_vl_V, αl_rhol_wl_V, αg_rhog_ug_V, αg_rhog_vg_V, αg_rhog_wg_V, αl_rhol_El_V, αg_rhog_Eg_V}**. 
Is that right?

**Reply by licharlot** [reply]
Yes that's correct.

**Reply by SomnusYu** [reply]
Thank you very much!
